"ninja -C builddir test" fails on appveyor with 0.48.0

```
ninja -C builddir test
ninja: Entering directory `builddir'
[0/1] Running all tests.
usage: meson [-h]
             [--buildtype {plain,debug,debugoptimized,release,minsize,custom}]
             [--strip] [--unity {on,off,subprojects}] [--prefix PREFIX]
             [--libdir LIBDIR] [--libexecdir LIBEXECDIR] [--bindir BINDIR]
             [--sbindir SBINDIR] [--includedir INCLUDEDIR] [--datadir DATADIR]
             [--mandir MANDIR] [--infodir INFODIR] [--localedir LOCALEDIR]
             [--sysconfdir SYSCONFDIR] [--localstatedir LOCALSTATEDIR]
             [--sharedstatedir SHAREDSTATEDIR] [--werror]
             [--warnlevel {1,2,3}] [--layout {mirror,flat}]
             [--default-library {shared,static,both}]
             [--backend {ninja,vs,vs2010,vs2015,vs2017,xcode}] [--stdsplit]
             [--errorlogs] [--install-umask INSTALL_UMASK]
             [--auto-features {enabled,disabled,auto}]
             [--optimization {0,g,1,2,3,s}] [--debug] [-D option]
             [--cross-file CROSS_FILE] [-v]
             [--wrap-mode {default,nofallback,nodownload,forcefallback}]
             [--fatal-meson-warnings] [--reconfigure]
             [builddir] [sourcedir]
meson: error: unrecognized arguments: -u --no-rebuild --print-errorlogs
FAILED: meson-test 
"C:\\python37\\Scripts\\meson" "-u" "test" "--no-rebuild" "--print-errorlogs"
ninja: build stopped: subcommand failed.
Command exited with code 1
```
The same configuration worked in 0.47.2.

Full build log at https://ci.appveyor.com/project/z3ntu/razer-test/build/job/8ak33l51nayn4a92

I got similar errors on Ubuntu in Travis CI.

(I'm working on adding Meson support to redshift)
```
$ if [ "$USE_MESON" -eq "1" ]; then ninja -v -C _build test; fi
ninja: Entering directory `_build'
[1/1] /home/travis/virtualenv/python3.6.3/bin/meson -u test --no-rebuild --print-errorlogs
usage: meson [-h]
             ...
             [builddir] [sourcedir]
meson: error: unrecognized arguments: -u --no-rebuild --print-errorlogs
FAILED: /home/travis/virtualenv/python3.6.3/bin/meson -u test --no-rebuild --print-errorlogs
ninja: build stopped: subcommand failed.
```

The `test` fails only when one of Meson's parent directories contains "python" (See `get_build_command()` in `environment.py`). In the example above, "/home/travis/virtualenv/**python**3.6.3/bin/meson" makes `get_build_command()` insert `-u` option.

Steps to reproduce the issue:
1. Rename one of Meson's parent directories to a name that contains "python"
2. Run Meson
3. Run `ninja -C _builddir_ test`

--- a/mesonbuild/environment.py
+++ b/mesonbuild/environment.py
@@ -375,7 +375,7 @@ class Environment:
 
     def get_build_command(self, unbuffered=False):
         cmd = mesonlib.meson_command[:]
-        if unbuffered and 'python' in cmd[0]:
+        if unbuffered and 'python' in os.path.basename(cmd[0]):
             cmd.insert(1, '-u')
         return cmd
 

